pyvrp/search/neighbourhood.py

```diff
@@ -157,30 +157,52 @@ def _compute_proximity(
     durations = data.duration_matrices()
     unique_edge_costs = {
         (
-            veh_type.unit_distance_cost,
-            veh_type.unit_duration_cost,
+            np.float32(veh_type.unit_distance_cost),
+            np.float32(veh_type.unit_duration_cost),
             veh_type.profile,
         )
         for veh_type in data.vehicle_types()
     }
 
     first, *rest = unique_edge_costs
     unit_dist, unit_dur, prof = first
-    edge_costs = unit_dist * distances[prof] + unit_dur * durations[prof]
-    for unit_dist, unit_dur, prof in rest:
-        mat = unit_dist * distances[prof] + unit_dur * durations[prof]
-        np.minimum(edge_costs, mat, out=edge_costs)
+    edge_costs = unit_dist * distances[prof].astype(np.float32)
+    edge_costs += unit_dur * durations[prof].astype(np.float32)
+    buf = np.empty_like(edge_costs)
 
-    # Minimum wait time and time warp of visiting j directly after i.
-    min_duration = np.minimum.reduce(durations)
-    min_wait = early[None, :] - min_duration - service[:, None] - late[:, None]
-    min_tw = early[:, None] + service[:, None] + min_duration - late[None, :]
```

Comment:
This is the most costly step (ignoring profiles), resulting in peak memory of 6x matrices of int64.